pkg/ast/pipesearch/searchHandler.go

```diff
@@ -527,41 +530,64 @@ func convertQueryCountToTotalResponse(qc *structs.QueryCount) interface{} {
 }
 
 /*
+Function designed to parse Time from a String input.
    Supports "now-[Num][Unit]"
    Num ==> any positive integer
    Unit ==> m(minutes), h(hours), d(days)
 */
 
-func parseAlphaNumTime(nowTs uint64, inp string, defValue uint64) uint64 {
+func parseAlphaNumTime(nowTs uint64, inp string, defValue uint64) (uint64, error) {
 
+	//removing spaces if any from input
 	sanTime := strings.ReplaceAll(inp, " ", "")
 
+	//to be returned as is if no filters are applied
 	if sanTime == "now" {
-		return nowTs
+		return nowTs, nil
 	}
 
+	//setting return value as default
 	retVal := defValue
 
+	//regex pattern for "now-[number]" and unit at the end which only accepts 'm', 'h' and 'd'
+	pattern := `^now-\d+[mhd]$`
+	r, patternError := regexp.Compile(pattern)
+	if patternError != nil {
+		log.Errorf("startEpoch/endEpoch Input received is invalid. Value probably modified")
+		return retVal, fmt.Errorf("startEpoch Input received is invalid.")
+	}
+
 	strln := len(sanTime)
-	if strln < 6 {
-		return retVal
+	patternMatch := r.MatchString(sanTime)
+	if !patternMatch {
+		//impossible case reached, return error
+		log.Errorf("startEpoch/endEpoch Input received is invalid. Value probably modified")
+		return retVal, fmt.Errorf("startEpoch Input received is invalid.")
 	}
```

Comment:
This is not an impossible case. This is what happens if the input from the user is invalid. So change/remove the comment. Also we don't know if this is for the `startEpoch` or `endEpoch`, and we shouldn't care, so for your error messages you can just say `Input received is invalid.` You can also remove `Value probably modified` (why do you have this?)